src/utils/AppDatabase.ts

```diff
@@ -27,14 +33,21 @@ export class AppDatabase {
             `);
         }
 
+        AppDatabase._initCount++;
         return appDb;
     }
 
+    /**
+     * Tracks the number of database initializations. Used to catch accidental re-initializations when the connection
+     * is not properly passed between screens.
+     */
+    private static _initCount: number = 0;
```

Comment:
Are we ever actually using this? Also, what's the advantage of this over a static instance with safe single init?

Reply:
Both honest questions, oops.

Reply:
For the first question: yup! It's being used by `UIScreen` when it calls the `init()`.

For the second question: it keeps `AppDatabase` instances from being stored in global state. I don't see too much of an issue that in some cases where you need to share a single connection per process, but some people see it as an anti-pattern.

Reply:
Excellent. Thanks!